### ISSUE TITLE
Missing Plugin : SplineRenderer

### ISSUE THREAD

**Issue by alx** [kwonoh/ImsvGraphVis#1]
Hi VIDILabs,

I've tried to launch your project (which looks fantastic) on a new install of Unreal Engine, but I've got this issue with a missing plugin:

> This project requires the 'SplineRenderer' plugin, which is not compatible with the current engine version.
> Would you like to disable it? You will no longer be able to open any assets created with it.

I've tried to copy the SplineRenderer folder located in the Plugins folder from the git repo to the Plugins folder in the Engine root path, but it doesn't seem to change anything.

I'm using Unreal Engine 4.17.2

Thanks and have a nice day,

Alex

**Comment by alx** [comment]
Investigation, seems related to Unreal Engine 4.17:

* "plugin not compatible" message appears for plugins that ship without binaries -
 https://answers.unrealengine.com/questions/697107/plugin-not-compatible-message-appears-for-plugin-t.html
* Inaccurate dialog box displayed when trying to open a project that contains a plugin with missing binaries - https://issues.unrealengine.com/issue/UE-49007

It seems to be related to missing binairies in the plugin, could you append this binary with SplineRenderer plugin ? I'm a noob with Unreal Engine, I don't understand if it's as easy as this.

Thanks,

Alex

**Comment by kwonoh** [comment]
Hi Alex, 

Could you please clean your project (using [Clean.bat](https://github.com/VIDILabs/ImsvGraphVis/blob/master/Clean.bat)) and test again with the newest commit (https://github.com/VIDILabs/ImsvGraphVis/commit/a1a7c131ee01b01c6d23174808df47c3bf565535)?

Thank you for your interest in this project.
Oh-Hyun

**Comment by alx** [comment]
Hi @kwonoh,

Thanks for your feedback.

I still have the same issue, here are some screenshot for you to view if I get the process right.

1. launch clean.bat
2. launch ImsvGraphVis Unreal Enginr Project
3. Message about missing plugin, asking to disable it, clicking no
4. Message about missing modules, asking to rebuild them, clicking yes
5. Error message telling me to rebuild from source manually

I'm not sure how to rebuild it from source, or if it's the solution, could you help ?

Thanks,

Alex

![imsvgraphvis_missing_plugin](https://user-images.githubusercontent.com/373/31333258-a43a95e8-ace9-11e7-98a3-1c78972d14b4.jpg)

![imsvgraphvis_missing_modules](https://user-images.githubusercontent.com/373/31333260-a43d0a80-ace9-11e7-8532-7b2a5eb636ba.jpg)

![imsvgraphvis_error](https://user-images.githubusercontent.com/373/31333259-a43b85ca-ace9-11e7-83ed-732ed4d5fcab.jpg)

**Comment by kwonoh** [comment]
Hi @alx,

For now, generate a Visual Studio project by right-clicking the `ImsvGraphVis.uproject` and launch the project from Visual Studio.

Thanks,
Oh-Hyun

**Comment by alx** [comment]
Thanks for your answer @kwonoh 

I've tried to generated a Visual Studio project like you said, but it's only generating a few file, and they don't seem to represent a Visual Studio project.

Too bad the 'show log' option from Unreal Engine dialog is disappearing so fast, there might be more information inside it.

Sorry to bother you with all this, but I'd really like to launch your project. I'm myself working on a graph vis web solution, and VR imersion is poping-up sometime. Feel free to tell me if you've no time for this.

![Files generated by 'generate Visual Studio project'](https://user-images.githubusercontent.com/373/31428506-3d899cb2-ae6b-11e7-9c75-548e01104cff.jpg)

**Comment by kwonoh** [comment]
Hi @alx,

You should find `ImsvGraphVis.sln` in the root directory, where the `ImsvGraphVis.uproject` is at. 
If there is no `ImsvGraphVis.sln`, please upload the log files in the `/Saved/Logs` directory.
I am happy to help you, so keep me posted :)

Thanks, 
Oh-Hyun

**Comment by alx** [comment]
Thnkas for pin-pointing me the location of log files, I've found what was missing:

> ERROR: Windows SDK v8.1 must be installed in order to build this target.

Found on https://developer.microsoft.com/en-us/windows/downloads/windows-8-1-sdk

I was able to open the project in Visual Studio, compile it, and open the editor window.
Then, the game seems to launch, but I've now some issues with the Vive as I've re-install the computer recently.

I think this issue can be closed, I'll let you know about future tests :)

Thanks for your help!
